### PR TITLE
#219 Added style in the error page

### PR DESCRIPTION
**Pull request by Nimanita**
Please suggest if any improvement needed.

### CODE REVIEW

**Comment by preetkaran20** [comment]
Hi @Nimanita ,

can you please attach screenshot of the UI for reference.

thanks,
Karan

File: src/main/resources/static/sasan.css [hunk 1]
```diff
@@ -1,6 +1,24 @@
...
+    margin-bottom: 0px;
+}
+p
+{
+    margin-top: 0px;
+    padding-top: 0px;
+    font-family: 'Secular One', sans-serif;
+    text-align: center;
+}
+.p1{
```
Comment (preetkaran20) [inline src/main/resources/static/sasan.css:21]: can you please give some meaningful name to this class.

File: src/main/resources/static/sasan.css [hunk 2]
```diff
@@ -1,6 +1,24 @@
...
+    justify-content: center;
     color: brown;
+}
+
+.sademoji
+{
+   font-size: 30px;
+    margin-bottom: 0px;
+}
+p
```
Comment (preetkaran20) [inline src/main/resources/static/sasan.css:14]: same here.

**Review by preetkaran20 — COMMENTED** [review]
Also can you please rename these pages to error.html and error.css.

thanks,
Karan

File: src/main/resources/static/sasan.html [hunk 3]
```diff
@@ -1,3 +1,8 @@
+<link href="https://fonts.googleapis.com/css2?family=Secular+One&display=swap" rel="stylesheet">
+<link rel="stylesheet" href="sasan.css" >
```
Comment (preetkaran20) [inline src/main/resources/static/sasan.html:2]: you need to change css file name here too and also need to change the sasan.html to error.html.

File: src/main/resources/static/sasan.html [hunk 4]
```diff
@@ -1,3 +1,8 @@
...
 <div id="content"> 
-    UserInterface for this Vulnerability level is still in-progress. Please use API interface for learning/testing this vulnerability. 
-</div>
+    <p class ="sademoji">&#128577;</p>
```
Comment (preetkaran20) [inline src/main/resources/static/sasan.html:4]: Great !!!

File: src/main/resources/static/sasan.html (hunk 3 above)
Comment (Nimanita) [reply src/main/resources/static/sasan.html:2]: Yes I will make all the recommended changes and will also attach the screenshot .Thank you

**Comment by Nimanita** [comment]
I have finished all the recommended changes and here attaching the screenshot of error page .

Thankyou

**Comment by preetkaran20** [comment]
Hi @Nimanita ,

can you please launch the VulnerableApp and send me the image of the level which is not having the UI which shows the html properly.

thanks,
Karan

**Comment by Nimanita** [comment]
I tried launching app using eclipde ide but it is showing some errors.I am attaching the screenshot of it.I have downloaded vulnerable app-1.0.0 folder and imported into eclipse.I followed all the steps in documentation buts its still not running

**Comment by preetkaran20** [comment]
> I tried launching app using eclipde ide but it is showing some errors.I am attaching the screenshot of it.I have downloaded vulnerable app-1.0.0 folder and imported into eclipse.I followed all the steps in documentation buts its still not running
> 

I think eclipse build is having some issues. please try building it in eclipse using gradle. may be try running eclipse under "sudo"/Super user permissions.

**Comment by Nimanita** [comment]
> > I tried launching app using eclipde ide but it is showing some errors.I am attaching the screenshot of it.I have downloaded vulnerable app-1.0.0 folder and imported into eclipse.I followed all the steps in documentation buts its still not running
> > 
> 
> I think eclipse build is having some issues. please try building it in eclipse using gradle. may be try running eclipse under "sudo"/Super user permissions.

I again configured graddle properly.This time application started running but I was not able to see api.what should I do?Attaching screenshots.

**Comment by preetkaran20** [comment]
Hi @Nimanita ,

Didn't got your question but i think you need to launch the application and then go to "http://localhost:9090"

**Comment by Nimanita** [comment]
> Hi @Nimanita ,
> 
> Didn't got your question but i think you need to launch the application and then go to "http://localhost:9090"

App launched.Attaching screenshot of change error page.The app works fine when page is named 'sasan' but when I changed its named to 'error' it shows some error.I don't know java spring so I won't be able to change the name of files  .

**Comment by preetkaran20** [comment]
> > Hi @Nimanita ,
> > Didn't got your question but i think you need to launch the application and then go to "http://localhost:9090"
> 
> App launched.Attaching screenshot of change error page.The app works fine when page is named 'sasan' but when I changed its named to 'error' it shows some error.I don't know java spring so I won't be able to change the name of files .

Thanks for the change. Regarding the issue of `sasan` and `error`, please go to vulnerableApp.js and change sasan to error as shown below.
image
this will fix the issue.

One more thing it would be great if we can remove show help and hide help buttons on error page. (For this you need to look at VulnerableApp.js)

thanks,
Karan

**Comment by Nimanita** [comment]
> > > Hi @Nimanita ,
> > > Didn't got your question but i think you need to launch the application and then go to "http://localhost:9090"
> > 
> > 
> > App launched.Attaching screenshot of change error page.The app works fine when page is named 'sasan' but when I changed its named to 'error' it shows some error.I don't know java spring so I won't be able to change the name of files .
> 
> Thanks for the change. Regarding the issue of `sasan` and `error`, please go to vulnerableApp.js and change sasan to error as shown below.
> image
> this will fix the issue.
> 
> One more thing it would be great if we can remove show help and hide help buttons on error page. (For this you need to look at VulnerableApp.js)
> 
> thanks,
> Karan

Yes I will do it.Thank you

**Comment by Nimanita** [comment]
> > > > Hi @Nimanita ,
> > > > Didn't got your question but i think you need to launch the application and then go to "http://localhost:9090"
> > > 
> > > 
> > > App launched.Attaching screenshot of change error page.The app works fine when page is named 'sasan' but when I changed its named to 'error' it shows some error.I don't know java spring so I won't be able to change the name of files .
> > 
> > 
> > Thanks for the change. Regarding the issue of `sasan` and `error`, please go to vulnerableApp.js and change sasan to error as shown below.
> > image
> > this will fix the issue.
> > One more thing it would be great if we can remove show help and hide help buttons on error page. (For this you need to look at VulnerableApp.js)
> > thanks,
> > Karan
> 
> Yes I will do it.Thank you

All tasks are completed.Attaching the screenshot.

**Comment by Nimanita** [comment]
While adding vulnerable file to repo.I got warning "warning: LF will be replaced by CRLF in src/main/resources/static/vulnerableApp.js.".Has my one check failed due to this reason?if yes than what should I do?

**Comment by preetkaran20** [comment]
> While adding vulnerable file to repo.I got warning "warning: LF will be replaced by CRLF in src/main/resources/static/vulnerableApp.js.".Has my one check failed due to this reason?if yes than what should I do?

Please run  './gradlew spotlessApply' i think this will fix the issue.

**Review by preetkaran20 — APPROVED** [review]
Please do the formatting and also fix the class name.
Other things LGTM.

thanks,
Karan

**Comment by Nimanita** [comment]
how should I run ./gradlew spotlessApply?

**Comment by preetkaran20** [comment]
> how should I run ./gradlew spotlessApply?

You need to run in the project i.e. from vulnerableapp directory

**Comment by preetkaran20** [comment]
Hi @Nimanita ,

Why are you deleting VulnerableApp.js ?

**Comment by preetkaran20** [comment]
If you are not able to run SpotlessApply from command line then i think you can run from eclipse too. 
under gradle tasks you should find one task called spotless.
image

**Comment by Nimanita** [comment]
> Hi @Nimanita ,
> 
> Why are you deleting VulnerableApp.js ?

Because problem arises after adding that file.

**Comment by preetkaran20** [comment]
> > Hi @Nimanita ,
> > Why are you deleting VulnerableApp.js ?
> 
> Because problem arises after adding that file.

yes because format of the code which you added is not proper and format will be fixed with SpotlessApply plugin. In case you are still facing issues with running gradle task please let me know, I will do that from my side.

thanks,
Karan

**Comment by Nimanita** [comment]
> > > Hi @Nimanita ,
> > > Why are you deleting VulnerableApp.js ?
> > 
> > 
> > Because problem arises after adding that file.
> 
> yes because format of the code which you added is not proper and format will be fixed with SpotlessApply plugin. In case you are still facing issues with running gradle task please let me know, I will do that from my side.
> 
> thanks,
> Karan

yes I am trying

**Comment by Nimanita** [comment]
I tried running gradlew  spotlessApply in command prompt but it shows following error

**Comment by Nimanita** [comment]
My build is successful but still checks are not passed.

**Comment by preetkaran20** [comment]
> My build is successful but still checks are not passed.
> 

Merging now. Will take care of it on my side.

thanks,
Karan

**Comment by preetkaran20** [comment]
Thanks a lot for the contribution, LGTM !!!.

**Comment by Nimanita** [comment]
Thank you

**Pull request merged**